brainscore_language/benchmarks/syntaxgym/__init__.py

```diff
@@ -0,0 +1,193 @@
+        """
+        candidate.start_behavioral_task(task=ArtificialSubject.Task.reading_times)
+        region_totals = []
+
+        # SyntaxGym logic wrapper around digest_text
+        for item_num, item in enumerate(self.suite.items):
+            region_totals_i = {}
+
+            for condition in item["conditions"]:
+                text_parts = [region["content"] for region in condition["regions"]]
+                surprisals = candidate.digest_text(text_parts)['behavior']
+                for i, region in enumerate(self.suite.region_names):
+                    surprisal_i = surprisals[i].values
+                    if np.isnan(surprisal_i):
+                        surprisal_i = 0.
+                    region_totals_i[condition["condition_name"], i + 1] = surprisal_i
+
+            region_totals.append(region_totals_i)
+        
+        return region_totals
+
+    def evaluate_predictions(self, region_totals: List[Dict[Tuple[str, int], float]]
+                             ) -> List[List[bool]]:
+        """
+        Compute prediction results for each item.
+        """
+        prediction_results = []
+        for item_region_totals in region_totals:
+            prediction_results.append([
+                pred.apply_prediction_formula(item_region_totals)
+                for pred in self.suite.predictions
+            ])
+
+        return prediction_results
+
+    def __call__(self, candidate: ArtificialSubject)-> Score:
+        region_totals = self.get_region_totals(candidate)
+        prediction_results = self.evaluate_predictions(region_totals)
+
+        # Compute conjunction of all predictions within-item.
+        conj_predictions = np.array(prediction_results).all(axis=1)
+        targets = [True] * len(conj_predictions)
+        score = self.metric(conj_predictions, targets)
+
+        return score
+
+
+# Map from test suite tag to high-level circuit.
+circuits = {
+    "Licensing": ["npi", "reflexive"],
+    "Long-Distance Dependencies": ["fgd", "cleft"],
+    "Agreement": ["number"],
+    "Garden-Path Effects": ["npz", "mvrr"],
+    "Gross Syntactic State": ["subordination"],
+    "Center Embedding": ["center"],
+}
+
+tag_to_circuit = {tag: circuit
+                  for circuit, tags in circuits.items()
+                  for tag in tags}
+
+
+benchmark_registry["syntaxgym-2020"] = SyntaxGym2020
+
+with open(Path(__file__).parent / 'test_suites.json') as json_file:
+    test_suite_dict = json.load(json_file)
+# Define a suite hierarchy: all suites are nested beneath the SyntaxGym benchmark;
+# with intervening "circuit" structure.
+for circuit, suite_tags in circuits.items():
+    # Create a benchmark with matching suites.
+    matching_suites = [suite_name for suite_name in test_suite_dict.keys()
+                       if suite_name.startswith(tuple(suite_tags))]
+    benchmark_registry[f"syntaxgym-2020/{circuit}"] = lambda: SyntaxGymTSE(matching_suites)
```

Comment:
@kvfairchild as far as I can tell, this will not play straight-forwardly with the plugin identifier matching via code parsing. How should we proceed here?